src/client/pmix_client.c

```diff
@@ -964,7 +964,7 @@ static void _commitfn(int sd, short args, void *cbdata)
      * that we contributed whatever we had */
     PMIX_PTL_SEND_RECV(rc, &pmix_client_globals.myserver, msgout,
                        wait_cbfunc, (void*)&cb->active);
-    if (PMIX_SUCCESS != rc) {
+    if (PMIX_SUCCESS == rc) {
```

Comment:
I looked at this some more and convinced myself that your original fix was the correct one! I'll update my branch accordingly.